openedx/core/djangoapps/xblock/api.py

```diff
@@ -123,7 +132,7 @@ def get_block_metadata(block, includes=()):
     data = {
         "block_id": str(block.scope_ids.usage_id),
         "block_type": block.scope_ids.block_type,
-        "display_name": get_block_display_name(block),
+        "display_name": block.display_name,
```

Comment:
This seems like a change - the fallback behavior will be missing if the block hasn't defined a display_name, won't it?

Reply:
Yeah. I think what happened is that I was originally focusing on the `block_or_key` aspect of it, got rid of the ability for it to be a key, and somewhere through the refactorings dropped the fallback behavior in `get_block_display_name`. But yeah, it should fallback, and in the current code it's not.